src/Main.tsx

```diff
@@ -9,63 +9,88 @@ import {
 } from '@material-ui/core'
 import {ManagedQuestionJSON} from './types'
 import queryString from 'query-string'
+import { useWindowSize } from "@reach/window-size";
 
 
 interface Props {
   managedQuestions: ManagedQuestionJSON;
   allTags: string[];
   search: string;
-  setSearch: Function;
+  allCategories: string[];
 }
 
 const Main = (props: Props) => {
   const history = useHistory()
+  const windowSize = useWindowSize()
 
   const params = queryString.parse(history.location.search)
-  if (typeof params.search !== 'string') {
-    params.search = ''
-  }
-  if (typeof params.integration !== 'string') {
-    params.integration = ''
+  if (typeof params.integrations !== 'string') {
+    params.integrations = ''
```

Comment:
These assignments should not happen. You want to just have a default. Make a new variable such as "integrations" and set it to params.integration or an empty string. Refactor some of this with that in mind.

Reply:
You also are doing a lot of checks for an empty string in this function. See if you can think of some ways to consolidate these.